### DISCUSSION TITLE
Import character creation Backgrounds from DDA

### DISCUSSION THREAD

**Discussion by Satellite-SLickers** [cataclysmbn/Cataclysm-BN#4923] [General]
### Is your feature request related to a problem? Please describe.

Backgrounds in the DDA are a wonderful way to give more character creation options, and i do miss them while playing Bright Nights

### Describe the solution you'd like

Import the code for backgrounds into Bright Nights

**Comment by chaosvolt** [comment]
You mean porting them from DDA I assume, which isn't really the "main" branch for us given we've long since split off from this repo being a fork of DDA's repo, code-wise.

As for the idea itself, the problem there is what does it accomplish mechanically, that's both compatible with BN's game design and that can't already be accomplished using existing methods?
1. A lot of background effects just grant a profession grant, wholly redundant as that can just be made a trait you purchase the normal way.
2. Proficiencies are right out, we aren't at present planning to add proficiency to BN because as-implemented their effects are primarily negatives that get reverted to baseline when the proficiency is, the player doesn't gain anything like faster-than-baseline crafting of recipes or unlocking bonus recipes for having the proficiency in chargen. Weakpoint mechanics are maybe the only genuine bonus if I recall, but those started as a bandaid for flaws in archery mechanics that aren't relevant to BN.
3. Skill bonuses are...well, that's what the skill menu of chargen is for. Packages that count as bonus traits that grant you some skill bonuses could be interesting but not really that interesting, plus BN still actually uses a point buy system and tries to at least try to make its point costs feel balanced (DDA uses what's basically freeform but it tries to predict how easy you've made things as its default), so balancing skill packages against the skill submenu and against profession skill bonuses would make something that's quite fiddly and subjective even more complicated.
4. Bonus starting items reflective of "you brought these items with you" might be a good use, but I don't know if DDA actually uses this as a bonus for any of its lifestyle traits. Moreover, traits already can grant bonus items (if I recall, Cannibal starting with To Serve Man is presently our only JSON example of this), so there's zero need to port backgrounds just for this.
5. The only other existing chargen thing that traits can't really do for you are addictions, which are presently only available at the profession level. If you really want to roleplay as an alcoholic without picking a profession that starts with it then I guess that's a viable addition, but I see no reason why we'd need backgrounds just for that as opposed to adding support for traits that grant the relevant addiction on game start. Balance-wise they're also really meaningless, since addictions currently both build up and decay super-fact, and it's an example of a drawback that logically should give you a permanent headstart in chargen with bonus points but has no permanent effects on the playthrough.

**Comment by Ignaramico** [comment]
i always thought the professions (or the starting scenarios)were the backgrounds

**Comment by Satellite-SLickers** [comment]
backgrounds are an aditional system that functions similarly to professions (being collections of traits, professions, abilties, and gear) that have the added quirk of non exclusivity with each other, which helps add in a lot more possibility, personally i feel like the best example of what they can be used for is in dda's magiclism mod, which uses the framework too add in various "species". and i feel like if that framework was added then a lot more stuff would be done

**Comment by chaosvolt** [comment]
Again, you haven't told me anything useful in terms of what porting this system actually does that we can't just use profession traits for. This seems like it'd be better off moved to a discussion at this rate, as well.

**Comment by Zireael07** [comment]
Backgrounds can do much more than profession traits do. For one thing, you can have the same profession work with any of several backgrounds.

**Comment by chaosvolt** [comment]
_Such as?_ Neither of you have given any actual examples. Like, when I say profession traits I'm not _just_ thinking about traits locked to a specific profession. Refer back to my very first response here, and consider the martial art traits for example. Those are traits that aren't locked to a specific profession, instead any profession can just buy them in chargen and start with a martial art. If you want your skater boy character to start as a HEMA nerd for example, you pick the relevant skating-related profession then pick the melee weapon martial arts trait.

Again, I'd like a clear answer here: what, **_SPECIFICALLY_**, do we gain from bothering with porting this feature over that could not simply be replicated by tying the benefit to a purchasable trait? So far every apparent benefit to backgrounds is completely and entirely redundant given what investing points in skills and traits allow.

Like I said earlier, the only things we're missing here as far as I can see are proficiencies, which are not planned to be added anytime soon, and addictions which would be far less effort to just code support for purchasable starting traits that grant you a level of addiction.

**Comment by Zireael07** [comment]
The biggest benefit to me is the ability to easily create many different characters w/o poring over huge lists of traits. And without the feeling of "oh if I take this trait I wasted my trait on a roleplay thing that isn't giving me poWEr"

Granted I am more of a roleplayer than a power player

**Comment by chaosvolt** [comment]
You want us to bother with setting aside the time and hassle to port an entirely new menu, that every player will have to sift through, because you want to sift through a huge list of backgrounds instead of a huge list of traits. There's also a flaw with your last reasoning:
```JSON
  {
    "type": "profession",
    "subtype": "hobby",
    "id": "parkour",
    "name": "Parkour",
    "description": "You've practiced parkour for many years, and made the world your playground.  It wouldn't be a lie to say that running is your life.  Which is good, because now that the end has come, you're running for your life.",
    "points": 2,
    "proficiencies": [ "prof_parkour", "prof_athlete_basic" ],
    "skills": [ { "level": 1, "name": "swimming" } ]
  },
```

Backgrounds ALSO have point costs. They combine traits and skill bonuses sure, so they could possibly save you on points in the long run, but the only reason you aren't having the same "is this worth it?" question in DDA is because survivor mode is literally just rebranded freeform. Since BN still uses point buy, even if we dealt with the code issues of porting an entire new section of UI over that does not grant us anything useful you can't already do with buyable traits or skills, we'd then have to figure out what point costs are balanced for them because we still actually bother with point cost balance. You'd still have that problem whether we make them buyable traits or port a whole new system just for them.

**Comment by ghost** [comment]
The only thing I'd say we lack right now is choosing starting items independent of profession. I think if we were going to do anything we'd want a way for players to pick items. I can't comment on what form that should take because the thermonuclear option is to make a whole new menu, whilst traits seem to be capable of adding items (or maybe should have that function if its broken or not existent).

Everything else exists within the current chargen.

Edit: It is confirmed on the Discord that traits allow you to add starter items. I will probably do something with this knowledge.

**Comment by ghost** [comment]
> backgrounds are an aditional system that functions similarly to professions (being collections of traits, professions, abilties, and gear) that have the added quirk of non exclusivity with each other, which helps add in a lot more possibility, personally i feel like the best example of what they can be used for is in dda's magiclism mod, which uses the framework too add in various "species". and i feel like if that framework was added then a lot more stuff would be done

Background does not help facilitate species work. Traits are how you create species and I've done that already in Essence 2200. If you want species you have to painstakingly block every unfitting trait in the json, among other things like assign effects. If you want species to be easier or better you have to improve the traits system, those are two separate systems.

Two examples of improving traits for species: Allow a trait to forcibly block mutations as a whole with a simple new flag, or block entire categories such as fish or ursine. And allow a mutation to automatically add relevant supporting mutations at start (example if I add 5 mutations related to a species a player has to pick all 5, or I have to bundle all 5 into one mutation, both are not ideal due to flavor reasons.)

Bonus example of improving traits for species: A way to actually block the installation of CBM's on mutations. Trust me when I say that ""no_cbm_on_bp"" does absolutely nothing unless you play with CBM slots AND you ensure all CBM's have slots including every mod, this is not ideal to begin with as a global block is necessary.
